src/swap/criptointercambio.js

```diff
@@ -0,0 +1,488 @@
+    ) {
+      throw new SwapCurrencyError(
+        swapInfo,
+        request.fromCurrencyCode,
+        request.toCurrencyCode
+      )
+    }
+    throw new Error('Criptointercambio error: ' + JSON.stringify(reply.error))
+  }
+}
+
+export function makeCriptointercambioPlugin(
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const { initOptions, io, log } = opts
+  const { fetchCors = io.fetch } = io
+
+  if (initOptions.apiKey == null || initOptions.secret == null) {
+    throw new Error('No Criptointercambio apiKey or secret provided.')
+  }
+  const { apiKey } = initOptions
+  const secret = parseUtf8(initOptions.secret)
+
+  async function call(json: any, promoCode?: string) {
+    const body = JSON.stringify(json)
+    const sign = base16
+      .stringify(hmacSha512(parseUtf8(body), secret))
+      .toLowerCase()
+
+    const headers: { [header: string]: string } = {
+      'Content-Type': 'application/json',
+      'api-key': apiKey,
+      sign
+    }
+    if (promoCode != null) headers['X-Promo-Code'] = promoCode
+    const response = await fetchCors(uri, { method: 'POST', body, headers })
+
+    if (!response.ok) {
+      throw new Error(`Criptointercambio returned error code ${response.status}`)
+    }
+    return response.json()
+  }
+
+  const out: EdgeSwapPlugin = {
+    swapInfo,
+    async fetchSwapQuote(
+      request: EdgeSwapRequest,
+      userSettings: Object | void,
+      opts: { promoCode?: string }
+    ): Promise<EdgeSwapQuote> {
+      checkInvalidCodes(INVALID_CURRENCY_CODES, request, swapInfo)
+
+      const fixedPromise = this.getFixedQuote(request, userSettings, opts)
+      // FIXME: Estimated swaps are temporarily disabled
+      const fixedResult = await fixedPromise
+      return fixedResult
+    },
+
+    async getFixedQuote(
+      request: EdgeSwapRequest,
+      userSettings: Object | void,
+      opts: { promoCode?: string }
+    ): Promise<EdgeSwapQuote> {
+      const { promoCode } = opts
+      const [fromAddress, toAddress] = await Promise.all([
+        getAddress(request.fromWallet, request.fromCurrencyCode),
+        getAddress(request.toWallet, request.toCurrencyCode)
+      ])
+      const quoteAmount =
+        request.quoteFor === 'from'
+          ? await request.fromWallet.nativeToDenomination(
+            request.nativeAmount,
+            request.fromCurrencyCode
+          )
+          : await request.toWallet.nativeToDenomination(
+            request.nativeAmount,
+            request.toCurrencyCode
+          )
+
+      const { safeFromCurrencyCode, safeToCurrencyCode } = safeCurrencyCodes(
+        CURRENCY_CODE_TRANSCRIPTION,
+        request
+      )
+
+      const fixedRateQuoteResponse: FixedRateQuote = await call({
+        jsonrpc: '2.0',
+        id: 'one',
+        method: 'getFixRateForAmount',
+        params: {
+          from: safeFromCurrencyCode,
+          to: safeToCurrencyCode,
+          amountFrom: quoteAmount
```

Comment:
`quoteAmount` is not always the FROM amount. Need to check the `request.quoteFor` value and check if it's `from` or `to`. If you can't do quotes coming from `to` amounts, then you need to throw.